sdk/freestanding-execution-engine/src/main.rs

```diff
@@ -361,12 +311,12 @@ fn read_program_file(fname: &str) -> Vec<u8> {
     file.read_to_end(&mut contents).unwrap_or_else(|err| {
         eprintln!(
             "Cannot read WASM binary '{}' to completion.  Error '{}' returned.",
-            fname, err
+            file_path, err
         );
         exit(-1);
     });
 
-    contents
+    (Path::new(file_path).file_name().unwrap().to_str().unwrap().to_string(), contents)
```

Comment:
Can something be done about these `unwrap` functions?  Either turn them into `expect(...)` or manually handle them using e.g. `unwrap_or_else()` as in the function calls above.